### PR TITLE
Various fixes to queries/perl/*

### PR DESCRIPTION
**Pull request by leonerd**
* Recognise many more keywords
* Assign highlight groups to control keywords, not entire structures
* Add some folding rules

Verymuch a work-in-progress as many bigger fixes will require improvements in `tree-sitter-perl`, but this is a good start.

### CODE REVIEW

**Review by clason — APPROVED** [review]

File: queries/perl/highlights.scm [hunk 1]
```diff
@@ -1,85 +1,116 @@
...
+[ "sub" ] @keyword.function
+
+; Lots of builtin functions, except tree-sitter-perl doesn't emit most of
+;   these yet
+;[
+;  "print" "printf" "sprintf" "say"
+;  "push" "pop" "shift" "unshift" "splice"
+;  "exists" "delete" "keys" "values"
+;  "each"
+;] @function.builtin
```
Comment (theHamsta) [inline queries/perl/highlights.scm:38]: Isn't it possible to recognize them with `#any-of?`? I'd imagine that they are at least present in the tree as `identifier`s in call position.

Comment (leonerd) [reply queries/perl/highlights.scm:38]: Uh quite possibly? :) I have never seen/heard-of `#any-of` before so perhaps a reference link would be useful?

Comment (clason) [reply queries/perl/highlights.scm:38]: You can look at other queries (Python, cmake). The documentation is `:h lua-treesitter-predicates`.

Comment (leonerd) [reply queries/perl/highlights.scm:38]: Ah, I see. That'd add quite a lot. To go through the *full* set of core functions, would probably add about 200 names here. I'm happy to do that but perhaps in a separate commit, to keep this one moving.
What do people think?

Comment (theHamsta) [reply queries/perl/highlights.scm:38]: sounds good. I feel that the distinction built-in function vs regular one is not essential

**Pull request merged**